tests/manage/z_cluster/test_log_rotate.py

```diff
@@ -0,0 +1,190 @@
+from ocs_ci.ocs.cluster import ceph_health_check
+from ocs_ci.ocs.resources import pod
+from ocs_ci.ocs.exceptions import TimeoutExpiredError
+from ocs_ci.ocs.ocp import OCP
+from ocs_ci.ocs import constants
+from ocs_ci.framework.testlib import (
+    ManageTest,
+    tier2,
+    bugzilla,
+    skipif_ocs_version,
+    skipif_external_mode,
+    ignore_leftovers,
+    config,
+)
+
+log = logging.getLogger(__name__)
+
+
+@tier2
+@ignore_leftovers
+@bugzilla("2116416")
+@skipif_external_mode
+@skipif_ocs_version("<4.10")
+@pytest.mark.polarion_id("OCS-4684")
+class TestLogsRotate(ManageTest):
+    """
+    Test Logs Rotate
+
+    """
+
+    @pytest.fixture(autouse=True)
+    def teardown(self, request):
+        def finalizer():
+            log.info("Delete logCollector from storage cluster yaml file")
+            storagecluster_obj = OCP(
+                resource_name=constants.DEFAULT_CLUSTERNAME,
+                namespace=defaults.ROOK_CLUSTER_NAMESPACE,
+                kind=constants.STORAGECLUSTER,
+            )
+            params = '[{"op": "remove", "path": "/spec/logCollector"}]'
+            storagecluster_obj.patch(params=params, format_type="json")
+            log.info(
+                "It takes time for storagecluster to update after the edit command"
+            )
+            time.sleep(30)
+            log.info("Verify storagecluster on Ready state")
+            verify_storage_cluster()
+            ceph_health_check()
+
+        request.addfinalizer(finalizer)
+
+    def get_pod_obj_based_on_id(self, pod_type):
+        """
+        Get Pod Obj
+
+        Args:
+            pod_type (str): The type of pod [osd/mon/mgr/rgw/mds]
+
+        Returns:
+            POD Obj:
+        """
+        pod_objs = self.podtype_id[pod_type][0]()
+        for pod_obj in pod_objs:
+            if self.podtype_id[pod_type][1] == pod.get_ceph_daemon_id(pod_obj):
+                return pod_obj
+
+    def verify_new_log_created(self, pod_type):
+        """
+        Verify new log created on /var/log/ceph
+
+        Args:
+            pod_type (str): The type of pod [osd/mon/mgr/rgw/mds]
+
+        Returns:
+            bool: True if a new log created, otherwise False
+
+        """
+        pod_obj = self.get_pod_obj_based_on_id(pod_type)
+        output_cmd = pod_obj.exec_cmd_on_pod(command="ls -lh /var/log/ceph")
+        expected_string = (
+            self.podtype_id[pod_type][2]
+            if pod_type == "rgw"
+            else f"{self.podtype_id[pod_type][2]}{self.podtype_id[pod_type][1]}"
+        )
+        cnt_logs = len(re.findall(expected_string, output_cmd))
+        if cnt_logs != int(self.podtype_id[pod_type][3]) + 1:
+            log.info(output_cmd)
+            log.error(
+                f"pod_type:{pod_type} cnt_logs_before_fill_log:"
+                f"{self.podtype_id[pod_type][3]} cnt_logs_after_fill_log:{cnt_logs}"
+            )
+            return False
+        return True
+
+    def test_logs_rotate(self):
```

Comment:
should we make it test_rook_ceph_log_rotate? and also the test file name the same way?

Reply:
done